src/components/views/messages/TextualBody.tsx

```diff
@@ -565,7 +575,36 @@ export default class TextualBody extends React.Component<IBodyProps, IState> {
         }
         return <span className="mx_EventTile_pendingModeration">{`(${text})`}</span>;
     }
+    private async decryptEmotes(): Promise<void> {
+        const client = MatrixClientPeg.safeGet();
+        const room = client.getRoom(this.props.mxEvent.getRoomId());
+        const emotesEvent = room?.currentState.getStateEvents(EMOTES_STATE.name, "");
+        const rawEmotes = emotesEvent ? emotesEvent.getContent() || {} : {};
+        const decryptedemotes = new Map<string, string>();
+        let decryptedurl = "";
+        const isEnc = client?.isRoomEncrypted(this.props.mxEvent.getRoomId()!);
+        for (const shortcode in rawEmotes) {
+            if (isEnc) {
+                const blob = await decryptFile(rawEmotes[shortcode]);
+                decryptedurl = URL.createObjectURL(blob);
+            } else {
+                decryptedurl = mediaFromMxc(rawEmotes[shortcode]).srcHttp!;
+            }
 
+            decryptedemotes.set(
+                ":" + shortcode + ":",
+                "<img class='mx_Emote' title=':" +
+                    shortcode.replace(/[^a-zA-Z0-9_]/g, "") +
+                    ":' src='" +
+                    decryptedurl +
+                    "'/>",
+            );
```

Comment:
We shouldn't be building HTML like this, its open to XSS injections

Reply:
The shortcode is stripped of all non alphanumeric or _ characters with the regex. I chose this character set to mimic the shortcode character set in popular apps like discord.

Reply:
It might be fine today but is brittle. We should be using the correct tools here, like react components. We avoid building raw HTML as it raises risk and causes CodeQL and other tooling to panic.

Reply:
HTMLUtils already uses sanitized HTML as strings instead of converting it to React Components and then uses dangerouslysetinnerhtml. This utilizes that existing code so I am not adding anything that was not there.